src/main/java/com/emc/mongoose/storage/driver/pravega/kvs/PravegaKVSDriver.java

```diff
@@ -70,14 +69,141 @@
+                    throwUnchecked(e);
+                } catch (final Throwable cause) {
+                    LogUtil.exception(
+                            Level.WARN, cause, "{}: failed to create the scope \"{}\"", stepId, scopeName
+                    );
+                }
+            }
+            return new KVTCreateFunctionImpl(controller, scopeName);
+        }
+    }
+
+    @Value
+    final class KVTCreateFunctionImpl
+            implements KVTCreateFunction {
+
+        Controller controller;
+        String scopeName;
+
+        @Override
+        public final KeyValueTableConfiguration apply(final String kvtName) {
+            KeyValueTableConfiguration kvtConfig = KeyValueTableConfiguration.builder()
+                    .partitionCount(partitionCount)
+                    .build();
+            if (controlKVTFlag) {
+                try {
+                    val createKVTFuture = controller.createKeyValueTable(scopeName, kvtName, kvtConfig);
+                    if (createKVTFuture.get(controlApiTimeoutMillis, MILLISECONDS)) {
+                        Loggers.MSG.trace(
+                                "KVT \"{}/{}\" was created using the config: {}",
+                                scopeName,
+                                kvtName,
+                                kvtConfig);
+                    } else {
+                        Loggers.MSG.info(
+                                "KVT \"{}\" was not created, may be already existing before", kvtName);
+                        //TODO: once it is supported
+                        //scaleToPartitionCount(
+                        //        controller, controlApiTimeoutMillis, scopeName, kvtName, scalingPolicy);
+                    }
+                } catch (final InterruptedException e) {
+                    throwUnchecked(e);
+                } catch (final Throwable cause) {
+                    LogUtil.exception(
+                            Level.WARN, cause, "{}: failed to create the KVT \"{}\"", stepId, kvtName);
+                }
+            }
+            return kvtConfig;
+        }
+    }
+
+    @Value
+    final class KVTFactoryCreateFunctionImpl
+            implements KVTFactoryCreateFunction {
+
+        ClientConfig clientConfig;
+
+        @Override
+        public final KeyValueTableFactory apply(final String scopeName) {
+            return KeyValueTableFactory.withScope(scopeName, clientConfig);
+        }
+    }
+
+    @Value
+    final class KVTClientCreateFunctionImpl
+            implements KVTClientCreateFunction {
+
+        KeyValueTableFactory kvtFactory;
+
+        @Override
+        public final KeyValueTable apply(final String kvtName) {
+            val kvtConfig = KeyValueTableClientConfiguration.builder()
+                    .build();
+            return kvtFactory.forKeyValueTable(kvtName, new UTF8StringSerializer(),
+                    new DataItemSerializer(false, false), kvtConfig);
```

Comment:
why have you decided to use custom Serializer?

Reply:
Because UTF8StringSerializer serializes String into ByteBuffer (public ByteBuffer serialize(String value)), whereas we need to transform Item into ByteBuffer. Which internally stores data in ByteBuffer as well.